dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerCommandBuilder.cs

```diff
@@ -585,14 +601,19 @@ private static void AddParameter(this SqlCommand command, PropertyModel? propert
                 command.Parameters.Add(name, System.Data.SqlDbType.VarBinary).Value = DBNull.Value;
                 break;
             case null:
-            case ReadOnlyMemory<float> vector when vector.Length == 0:
                 command.Parameters.AddWithValue(name, DBNull.Value);
                 break;
             case byte[] buffer:
                 command.Parameters.Add(name, System.Data.SqlDbType.VarBinary).Value = buffer;
                 break;
             case ReadOnlyMemory<float> vector:
-                command.Parameters.AddWithValue(name, JsonSerializer.Serialize(vector));
+                command.Parameters.AddWithValue(name, JsonSerializer.Serialize(vector, SqlServerJsonSerializerContext.Default.ReadOnlyMemorySingle));
```

Comment:
Note that this is the NativeAOT-compatible way to perform JSON serialization. SqlServerJsonSerializerContext is  source-generated with logic that allows safely serializing/deserializing .NET types (in this case, `ReadOnlyMemory<float>` and `float[]`), and we use that here.